src/dos/dos.cpp

```diff
@@ -1250,11 +1294,20 @@ class DOS:public Module_base{
 		DOS_SetupMisc();							/* Some additional dos interrupts */
 		DOS_SDA(DOS_SDA_SEG,DOS_SDA_OFS).SetDrive(25); /* Else the next call gives a warning. */
 		DOS_SetDefaultDrive(25);
-	
+
 		dos.version.major=5;
 		dos.version.minor=0;
 		dos.direct_output=false;
 		dos.internal_output=false;
+
+		const Section_prop* section = static_cast<Section_prop*>(configuration);
+		char *args = const_cast<char *>(section->Get_string("ver"));
+		const char* word = StripWord(args);
+		const auto new_version = DOS_ParseVersion(word, args);
+		if (new_version.major || new_version.minor) {
+			dos.version.major = new_version.major;
+			dos.version.minor = new_version.minor;
```

Comment:
`dos.version = new_version`. You don't need to rewrite struct field-by-field :) And it will work automatically once `DOS_ParseVersion` will be able to parse `revision` as well (as far as I can tell revision is a single character, which was sometimes (rarely) appended to DOS version, like: `3.3a` or `4.01a`.

Reply:
Done.